packages/cli/src/commands/setup/ui/tasks/setup-component-library.js

```diff
@@ -0,0 +1,211 @@
+ * correctly if there are multiple uses.
+ * imports are added after the last redwoodjs import.
+ * moduleScopeLines are added after the last import.
+ *
+ * @param {Object} options Configure behavior
+ * @param {Object} options.wrapTag Configure tag-wrapping behavior
+ * @param {string} options.wrapTag.wrappedComponent The name of the component to wrap.
+ * @param {string} options.wrapTag.wrapperComponent The name of the component to with which to wrap.
+ * @param {Object|string} options.wrapTag.wrapperProps Properties to pass to the wrapper component.
+ * @param {Array} options.imports Import declarations to inject after the last redwoodjs import.
+ * @param {Array} options.moduleScopeLines Lines of code to inject after the last import statement.
+ */
+export function extendAppJS({
+  wrapTag: {
+    wrappedComponent = undefined,
+    wrapperComponent = undefined,
+    wrapperProps = undefined,
+    before = undefined,
+    after = undefined,
+  },
+  imports = [],
+  moduleScopeLines = [],
+}) {
+  const webAppPath = getPaths().web.app
+  let content = fs.readFileSync(webAppPath).toString().split('\n')
+
+  if (moduleScopeLines && moduleScopeLines.length) {
+    content.splice(
+      content.findLastIndex((l) => l.trimStart().startsWith('import')) + 1,
+      0,
+      '', // Empty string to add a newline when we .join('\n') below.
+      ...moduleScopeLines
+    )
+  }
+
+  if (imports && imports.length) {
+    content.splice(
+      content.findLastIndex((l) => l.includes('@redwoodjs')) + 1,
+      0,
+      '', // Empty string to add a newline when we .join('\n') below.
+      ...imports
+    )
+  }
+
+  if (wrappedComponent && wrapperComponent) {
+    wrapComponentWithComponent(content, wrappedComponent, {
+      component: wrapperComponent,
+      props: wrapperProps,
+      before,
+      after,
+    })
+  }
+
+  fs.writeFileSync(
+    webAppPath,
+    content.filter((e) => e !== undefined).join('\n')
+  )
+}
+
+/**
+ * Inject lines of code into an array of lines to wrap the specified component in a new component tag.
+ * Increases the indentation of newly-wrapped content by two spaces (one tab).
+ *
+ * @param {Array} content A JSX file split by newlines.
+ * @param {*} wrappedComponent The name of the component to wrap with a new tag.
+ * @param {Object} _anonymousObject Configure the wrapping tag.
+ * @param {string} _anonymousObject.component Name of the wrapping tag.
+ * @param {Object|string|undefined} _anonymousObject.props Properties to pass to the wrapper
+ * component.
+ * @param {string} _anonymousObject.before A string to prepend, at the same indentation, as the
+ * wrapper.
+ * @param {Object} _anonymousObject.after A string to append, at the same indentation, as the wrapper.
+ * @returns Nothing; modifies content in place.
+ */
+function wrapComponentWithComponent(
+  content,
+  wrappedComponent,
+  { component, props, before, after }
+) {
+  const findTagIndex = (regexString) => {
```

Comment:
Could you make this take a regex instead of a regex string? That would also help get rid of the Code scanning failures below

Reply:
Done 👍